dm/syncer/data_validator.go

```diff
@@ -954,7 +955,11 @@ func (v *DataValidator) processRowsEvent(header *replication.EventHeader, ev *re
 
 func (v *DataValidator) checkAndPersistCheckpointAndData(loc binlog.Location) error {
 	metaFlushInterval := v.cfg.ValidatorCfg.MetaFlushInterval.Duration
-	if time.Since(v.lastFlushTime) > metaFlushInterval {
+	needCutOver := v.cutOverLocation != nil && binlog.CompareLocation(*v.cutOverLocation, loc, v.cfg.EnableGTID) <= 0
```

Comment:
if `v.cutOverLocation` is nil, validator will never `persistCheckpointAndData`?

Reply:
sorry I make a mistake. The following condition should be `or`.

Reply:
Is this fixed?

Reply:
Yes. `&&` is revised to `||` now